packages/react/src/hooks/useAuthSession.ts

```diff
@@ -0,0 +1,70 @@
+"use client";
+
+import { useSyncExternalStore } from "react";
+import { useConfig } from "wagmi";
+import { getAuthSession } from "@alchemy/wagmi-core";
+import type { AuthSession } from "@alchemy/auth";
+import type { ConfigParameter } from "../types";
+
+export type UseAuthSessionParameters = ConfigParameter;
+
+export type UseAuthSessionReturnType = AuthSession | null;
+
+/**
+ * React hook that returns the current authentication session.
+ *
+ * **Advanced usage only.** For most use cases, use higher-level hooks instead.
+ * If you just need user info, use `useUser()` instead.
+ *
+ * @param {UseAuthSessionParameters} parameters - Configuration options for the hook
+ * @param {Config} [parameters.config] - Optional wagmi config override
+ * @returns {AuthSession | null} The current auth session, or null if not authenticated
+ */
+export function useAuthSession(
+  parameters: UseAuthSessionParameters = {},
+): UseAuthSessionReturnType {
+  const config = useConfig(parameters);
+
+  const authSession = useSyncExternalStore(
+    (onChange) => {
+      let disconnectUnsubscribe: (() => void) | undefined;
+
+      // Subscribe to account status changes
+      const unwatchAccount = config.subscribe(
+        (state) => state.status,
```

Comment:
just realized -- i don't think we currently emit a status change on alchemyAuth when a session is restored silently. can you add a "change" event?

Reply:
added a `"connect"` event when resuming session.